src/spag/main.rs

```diff
@@ -118,14 +157,83 @@ fn spag_history(args: &Args) {
-    let mut withs: HashMap<&str, &str> = HashMap::new();
-    for (k, v) in args.arg_key.iter().zip(args.arg_val.iter()) {
-        withs.insert(k, v);
+    if args.cmd_list {
+        spag_request_list(args);
+    } else if args.cmd_show {
+        spag_request_show(args);
+    } else {
+        spag_request_a_file(args);
+    }
+}
+
+fn spag_request_a_file(args: &Args) {
+    let endpoint = try_error!(get_endpoint(args));
+    let dir = try_error!(get_dir(args));
+
+    match request::load_request_file(&args.arg_file, &dir) {
+        Ok(y) => {
+            let method = get_string_from_yaml(&y, &["method"]);
+            let uri = get_string_from_yaml(&y, &["uri"]);
+
+            let body =
+                if !args.flag_data.is_empty() {
+                    args.flag_data.to_string()
+                } else {
+                    if let Some(&Yaml::String(ref b)) = env::get_nested_value(&y, &["body"]) {
+                        b.to_string()
+                    } else {
+                        String::new()
+                    }
+                };
+
+            // env::get_nested_value(&y, &["body"]).unwrap_or(default_string);
+            let headers = try_error!(get_headers(args, &y));
+
+            // join headers into single strings...which will be split immediately afterwards
+            let str_headers: Vec<String> = headers.iter()
+                .map(|(k, v)| format!("{}: {}", k, v))
+                .collect();
+            //println!("{:?}", str_headers);
+
+            let mut req = SpagRequest::new(request::method_from_str(&method), endpoint, uri);
+            try_error!(req.add_headers(str_headers.iter()));
+            req.set_body(body);
+            do_request(&req);
+        },
+        Err(msg) => { error!("{}", msg); }
+    }
+    //println!("key={:?} val={:?}", args.arg_key, args.arg_val);
+    //let mut withs: HashMap<&str, &str> = HashMap::new();
+    //for (k, v) in args.arg_key.iter().zip(args.arg_val.iter()) {
+    //    withs.insert(k, v);
+    //}
+    //let text = template::untemplate(&args.arg_file, &withs, true);
+    //println!("untemplated: {:?}", text);
+}
+
+fn spag_request_show(args: &Args) {
+    let dir = try_error!(get_dir(args));
+    let filename = try_error!(request::get_request_filename(&args.arg_file, &dir));
+    // TODO: read_file() panics. don't do that.
+    let contents = file::read_file(&filename);
+    println!("{}", contents);
+}
+
+fn spag_request_list(args: &Args) {
+    let dir = try_error!(get_dir(args));
+    let filenames = try_error!(file::walk_dir(&dir));
+    let mut yaml_files: Vec<&PathBuf> = filenames.iter()
+        .filter(|p| p.to_str().unwrap().ends_with(".yml"))
+        .collect();
+    yaml_files.sort();
+
+    let current_dir = try_error!(std::env::current_dir());
+    for file in yaml_files.iter() {
+        // relative_from() is unstable
+        println!("{}", file.relative_from(&current_dir).unwrap().to_str().unwrap());
```

Comment:
What version of Rust do you have? I have:

    $ rustc --version
    rustc 1.2.0-nightly (a9515698f 2015-06-20)

And this definitely works for me. Or `relative_from` may work on Linux and not OS X which would be super dumb.